### DISCUSSION TITLE
Discussion on Julia

### DISCUSSION THREAD

**Discussion by ViralBShah** [github/india#77] [Open Source]
Starting a thread here for Julia discussions.

**Comment by tsadarsh** [comment]
Sir, you mentioned that Python packages can be imported in Julia. So does this mean we can run any `.py` file using Julia?

p.s. The talk was really interesting and informative.

**Reply by ViralBShah** [reply]
See https://github.com/JuliaPy/PyCall.jl. Julia is not executing your python code. For example, you can use matplotlib for plotting in Julia. Here's an example from the PyCall readme:

```
using PyCall
math = pyimport("math")
math.sin(math.pi / 4) # returns ≈ 1/√2 = 0.70710678...
```

A matplotlib example:

```
plt = pyimport("matplotlib.pyplot")
x = range(0;stop=2*pi,length=1000); y = sin.(3*x + 4*cos.(2*x));
plt.plot(x, y, color="red", linewidth=2.0, linestyle="--")
plt.show()
```

**Reply by tsadarsh** [reply]
I looked up https://github.com/JuliaPy/PyCall.jl. I understand how arbitrary Python libraries are imported and functions are called. This is cool. 
What is more cool is using `py"""..."""` similar to Python's `eval` and `exec`. Excellent :sunglasses:
